homeassistant/components/tplink/sensor.py

```diff
@@ -5,131 +5,28 @@
 from dataclasses import dataclass
 from typing import cast
 
-from kasa import Device, Feature, SmartStrip
+from kasa import Device, Feature
 
-from homeassistant.components.sensor import (
-    SensorDeviceClass,
-    SensorEntity,
-    SensorEntityDescription,
-    SensorStateClass,
-)
-from homeassistant.const import (
-    ATTR_VOLTAGE,
-    UnitOfElectricCurrent,
-    UnitOfElectricPotential,
-    UnitOfEnergy,
-    UnitOfPower,
-)
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import TPLinkConfigEntry
-from .const import (
-    ATTR_CURRENT_A,
-    ATTR_CURRENT_POWER_W,
-    ATTR_TODAY_ENERGY_KWH,
-    ATTR_TOTAL_ENERGY_KWH,
-)
 from .coordinator import TPLinkDataUpdateCoordinator
 from .entity import (
     CoordinatedTPLinkEntity,
     _description_for_feature,
-    _entities_for_device,
+    _entities_for_device_and_its_children,
 )
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, kw_only=True)
 class TPLinkSensorEntityDescription(SensorEntityDescription):
     """Describes TPLink sensor entity."""
 
-    emeter_attr: str | None = None
     precision: int | None = None
```

Comment:
I think we don't need this anymore, as we are reading the precision already from the feature.

Reply:
Done in [`b53d643` (#117764)](https://github.com/home-assistant/core/pull/117764/commits/b53d6433ba36ecefd2487acdaf75c58a1e5173a0)